code.ts

```diff
@@ -98,20 +143,42 @@ async function startTimer(node: TextNode, seconds: number, template: string, sta
   await figma.loadFontAsync(node.fontName as FontName);
   activeTimer += 1;
 
+  console.log("Timer started / became active");
+
+  var timerID = activeTimer;
+  var keepItRunning = true;
   var secondsToGo = seconds;
-  while(secondsToGo > 0) {
-    var newText = fillUpTimeStringWithTempalte(secondsToInterval(secondsToGo), template);
-    if (startsWithTimer) {
-        newText = "Timer: " + newText;
+  var newText = "";
+
+  while (keepItRunning) {
+
+    // checking if reset was clicked by user and if so resetting all timers
+    if (reset) {
+      secondsToGo = seconds;
+      newText = fillUpTimeStringWithTemplate(secondsToInterval(secondsToGo), template);
+      newText = "Timer: " + newText;
+      node.characters = newText;
+      keepItRunning = false;
+    };
+
+    // checking if pause was NOT clicked
+    if (!pause) {
+      if (secondsToGo > 0) {
+        newText = fillUpTimeStringWithTemplate(secondsToInterval(secondsToGo), template);
+        if (startsWithTimer) {
+          newText = "Timer: " + newText;
+        }
+        node.characters = newText;
+        secondsToGo -= 1;
+      } else if (secondsToGo < 1) {
+        node.characters = "Done";
+      }
+      await delay(1000);
+    } else {
+      await delay(1000);
     }
```

Comment:
Looks like we don't need the else case here and we can add the delay after the if block